src/plugins/intel_gpu/src/graph/include/program_node.h

```diff
@@ -530,7 +530,7 @@ template <class PType>
 struct typed_program_node : public typed_program_node_base<PType> {
     using typed_program_node_base<PType>::typed_program_node_base;
 
-    program_node& input() const { return program_node::get_dependency(0); }
+    program_node& input(std::size_t index = 0) const { return program_node::get_dependency(index); }
```

Comment:
Is there any reason to modify? Better readability?

Reply:
This is for future refactoring.
We have a lot of boilerplate code that can be avoided.
For example, following code

```
namespace cldnn {

template <>
struct typed_program_node<reverse> : public typed_program_node_base<reverse> {
    using parent = typed_program_node_base<reverse>;

public:
    using parent::parent;

    program_node& input(size_t index = 0) const {
        return get_dependency(index);
    }
};

using reverse_node = typed_program_node<reverse>;

```
can be reduced to 

`using reverse_node = typed_program_node<reverse>;
`

the same was done for matrix_nms primitive